### PR TITLE
Solution to Bacteria Reproduction problem in C++

### PR DESCRIPTION
**Pull request by Kuljeet-123**
**Fixes issue:** #4584 

**Changes:**

added solution to [BACREP](https://www.codechef.com/OCT19A/problems/BACREP) in C++.

### CODE REVIEW

**Comment by Kuljeet-123** [comment]
@arnavb Please review it and merge it also.

File: code/online_challenges/src/codechef/BACREP/BACREP.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+#define ll long long
+#define dd double
+#define endl "\n"
+#define pb push_back
+#define all(v) v.begin(),v.end()
+#define mp make_pair
+#define fi first
+#define se second
+#define fo(i,n) for(int i=0;i<n;i++)
+#define fo1(i,n) for(int i=1;i<=n;i++)
```
Comment (arnavb) [inline code/online_challenges/src/codechef/BACREP/BACREP.cpp:14]: Don't use any of this.

File: code/online_challenges/src/codechef/BACREP/BACREP.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+ll mod=1000000007;
```
Comment (arnavb) [inline code/online_challenges/src/codechef/BACREP/BACREP.cpp:15]: `constexpr` for all constants.

File: code/online_challenges/src/codechef/BACREP/BACREP.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+const ll  N=500050;
+bool vis[N];
+vector <int> adj[N];
+ll depth[N]={0};
+vector <int> atd[N];
+ll maxdepth=1;
+ll par[N]={0};
```
Comment (arnavb) [inline code/online_challenges/src/codechef/BACREP/BACREP.cpp:22]: Anything that doesn't have to be global shouldn't be.

File: code/online_challenges/src/codechef/BACREP/BACREP.cpp [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+ll po(ll k ,ll n,ll ans,ll temp,ll ans1)
+{
+    if(n==0)return ans;
+    
+    while(temp<=n)
+    {
+        ans*=ans1;ans%=mod;ans1=ans1*ans1;ans1%=mod;n=n-temp;temp*=2;
```
Comment (arnavb) [inline code/online_challenges/src/codechef/BACREP/BACREP.cpp:30]: Reformat file with clang format and 4 spaces.

File: code/online_challenges/src/codechef/NEWSCH/NEWSCH.cpp [hunk 5]
```diff
@@ -0,0 +1,52 @@
+#include <iostream>
+
+#include <cstring>
+
```
Comment (arnavb) [inline code/online_challenges/src/codechef/NEWSCH/NEWSCH.cpp:4]: Once again, your prs seem to have combined.

**Comment by Kuljeet-123** [comment]
@arnavb Due to conflicts , I am closing this PR and will raise a new PR. Because I  forked the repo again

**Pull request closed without merging**